src/Map.jsx

```diff
@@ -74,7 +82,40 @@ class Map extends Component {
       { force: true },
     );
   }
+  getStopInfo(route, info) {
+    let sInfo = {};
+    if (!this.state.stopInfo.isFirstStpChosen()) {
+      sInfo = update(this.state.stopInfo, { firstStop: { $set: info } });
+      const station = this.findStationID(route, info);
+      console.log(`First Stop ID: ${station.sid}`);
+    } else if (this.isSameStop(info, this.state.stopInfo.firstStop)) {
+      sInfo = update(this.state.stopInfo, { firstStop: { $set: {} } });
+    } else if (!this.state.stopInfo.isSecondStpChosen()) {
+      sInfo = update(this.state.stopInfo, { secondStop: { $set: info } });
+      const station = this.findStationID(route, info);
+      console.log(`Second Stop ID: ${station.sid}`);
+    } else if (this.isSameStop(info, this.state.stopInfo.secondStop)) {
+      sInfo = update(this.state.stopInfo, { secondStop: { $set: {} } });
+    } else if (this.state.stopInfo.isFirstStpChosen() && this.state.stopInfo.isSecondStpChosen()) {
+      const station = this.findStationID(route, info);
+      console.log(`First Stop ID: ${station.sid}`);
+      sInfo = update(this.state.stopInfo, { firstStop: { $set: info }, secondStop: { $set: {} } });
+    }
+    this.setState({ stopInfo: sInfo });
+  }
 
+  isSameStop(frstStop, scndStop) {
+    console.log(this);
+    const frstStpPosition = frstStop.object.position;
+    const scndStpPosition = scndStop.object.position;
+    return frstStpPosition[0] === scndStpPosition[0] && frstStpPosition[1] === scndStpPosition[1];
+  }
+  findStationID(route, stopInfo) {
+    console.log(this);
+    const stopLoc = stopInfo.object.position;
+    const station = route.stops.filter(c => c.lon === stopLoc[0] && c.lat === stopLoc[1]);
```

Comment:
Right now, I need to filter out an entire array to get one value back. (the station SID)

I'm matching the stops with the given lon/lat of the one that was clicked so I can get the station ID

Ideally, we'd get that info when the click gets bubbled up, but I didn't see the station ID; just the coordinates.

Reply:
How well does this work in practice? We could use turf.js to see whether those coordinates are within a stop radius? (we could add another helper file for that, as we don't really have a place for utils yet)

Reply:
@EddyIonescu the coordinates are accurate. A noted in your other comment, [objectName].object.position returns the exact coordinates of the bus stop; simply referencing [objectName].lngLat returns where the user click on the map. 
This works because we have the exact coordinates of the stop. It's just inefficient because we're looping through until we find it and we aren't even able to break out of the loop one we find it (using filter You can break out of a standard for loop easily)

Reply:
true - selecting the stop will give us the same stop coordinates that we provided.

Since we're only getting one item, couldn't we use `first()` instead of `filter()`?